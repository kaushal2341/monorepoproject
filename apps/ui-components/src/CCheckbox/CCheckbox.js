import React, {memo} from 'react';
import Form from 'react-bootstrap/Form';
import PropTypes from 'prop-types';

const CCheckbox = props => {
    const {
        id,
        label,
        ref,
        children,
        className,
        custom,
        disabled,
        feedback,
        inline,
        isInvalid,
        isValid,
        title,
        type,
        bsPrefix,
        onChange
    } = props;

    return (
        <Form.Check
            test-id="c-checkbox"
            id={label.concat('_').concat(id)}
            label={label}
            type={type}
            _ref={ref}
            children={children}
            className={className}
            custom={custom}
            disabled={disabled}
            feedback={feedback}
            inline={inline}
            isInvalid={isInvalid}
            isValid={isValid}
            title={title}
            bsPrefix={bsPrefix}
            onChange={onChange}
        />
    );
};

CCheckbox.defaultProps = {
    type: 'checkbox',
    onChange: () => {
    }
};

CCheckbox.propTypes = {
    id: PropTypes.string.isRequired, // must be unique
    label: PropTypes.oneOfType([
        PropTypes.string.isRequired,
        PropTypes.node
    ]),
    _ref: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.shape({current: PropTypes.instanceOf(Element)})
    ]),
    children: PropTypes.node,
    className: PropTypes.string,
    custom: PropTypes.bool,
    disabled: PropTypes.bool,
    feedback: PropTypes.node,
    inline: PropTypes.bool,
    isInvalid: PropTypes.bool,
    isValid: PropTypes.bool,
    title: PropTypes.string,
    bsPrefix: PropTypes.string,
    onChange: PropTypes.func.isRequired
};

export default memo(CCheckbox);
