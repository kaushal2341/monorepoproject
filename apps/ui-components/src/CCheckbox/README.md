# CCheckbox

`Checkbox` and `Radio` component made using react-bootstrap's Form.Check.

To use it, first

```jsx harmony
 import {CCheckbox} from "@cogent/ui-components";
```
Then to use it as checkbox pass type as `checkbox`,

```jsx harmony
 <CCheckbox 
     id="checkbox"
     label="Checkbox"
     type="checkbox"
     onChange={(e) => console.log(e)}
     title="checkbox1"
     inline={true}
     custom={true}
/>
```
And to use it as radio button, simply pass type as `radio`.

```jsx harmony
 <CCheckbox 
     id="radio"
     label="Radio Button"
     type="radio"
     onChange={(e) => console.log(e)}
     title="radio1"
     inline={true}
     custom={true}
/>
```

## CCheckbox props:
```
id: string - Id of each checkbox/radio. It must be UNIQUE.

label: string || node - Label to display.

_ref: ReactRef - The FormCheck ref will be forwarded to the underlying input element, 
                 which means it will be a DOM node, when resolved.

children: node - Provide a function child to manually handle the layout of the FormCheck's inner components.

className: string - Css class to customize the design.

custom: boolean - Use Bootstrap's custom form elements to replace the browser defaults.

disabled: boolean - default: false - To manually disable or enable the checkbox.

feedback: node - Message to display when the input is in a validation state.

inline: boolean- default: false - add class 'form-check-inline' to bring checkboxes on the same horizontal line.

isInvalid: boolean - default: false - Manually style the input as invalid, shows in red.

isValid: boolean - default: false - Manually style the input as valid, shows in green.

title: string - default: '' - Label's title property.

type: string('radio' | 'checkbox') -  default: 'checkbox' - The type of checkable.

bsPrefix: string - default:'form-check' - Change the underlying component
                    CSS base class name and modifier class names prefix.
                    This is an escape hatch for working with heavily
                    customized bootstrap css.

onChange: function - Function that handles checkbox/radio button change event.
```