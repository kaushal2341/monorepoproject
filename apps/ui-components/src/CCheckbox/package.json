{
  "name": "@cogent/c-checkbox",
  "version": "0.0.0",
  "main": "CCheckbox.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
