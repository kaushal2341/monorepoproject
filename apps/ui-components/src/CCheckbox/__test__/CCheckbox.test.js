import CCheckbox from '../CCheckbox';

expect.addSnapshotSerializer(enzymeSerializer);

describe('CCheckbox Tests', () => {
    let sWrapper, mockFunction;
    beforeEach(() => {
        mockFunction = jest.fn();
        sWrapper = shallow(<CCheckbox
            id="checkbox_1"
            label="checkItBaby"
            onChange={mockFunction}/>);
    });

    test('if CCheckbox is defined', () => {
        expect(sWrapper).toBeDefined();
    });

    test('if has one Form.Check checkbox element', () => {
        expect(sWrapper.find('[test-id="c-checkbox"]').length).toEqual(1);
    });

    test('if has type as checkbox', () => {
        expect(sWrapper.find('[test-id="c-checkbox"]').prop('type')).toEqual('checkbox');
    });

    test('if has label with value', () => {
        sWrapper.setProps({
            label: 'CheckItBaby'
        });
        expect(sWrapper.find('[test-id="c-checkbox"]').prop('label')).not.toBe("");
    });

    test('if checkbox has all required props', () => {
        let propRequired = [
            'test-id',
            'id',
            'label',
            'type',
            '_ref',
            'children',
            'className',
            'custom',
            'disabled',
            'feedback',
            'inline',
            'isInvalid',
            'isValid',
            'title',
            'bsPrefix',
            'onChange'
        ];

        let propsOfCheckbox = Object.keys(sWrapper.find('[test-id="c-checkbox"]').props());
        propRequired.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfCheckbox[i])
        ));
    });

    test('if onCheckHandler function is called when checked', () => {
        sWrapper.find('[test-id="c-checkbox"]').simulate('change');
        expect(mockFunction).toHaveBeenCalled();
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(sWrapper).toMatchSnapshot();
        })
    });

});