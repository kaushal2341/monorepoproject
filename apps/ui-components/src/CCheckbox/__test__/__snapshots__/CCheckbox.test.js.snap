// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCheckbox Tests Snapshot Testing should match snapshot 1`] = `
<FormCheck
  disabled={false}
  id="checkItBaby_checkbox_1"
  inline={false}
  isInvalid={false}
  isValid={false}
  label="checkItBaby"
  onChange={[MockFunction]}
  test-id="c-checkbox"
  title=""
  type="checkbox"
/>
`;
